src/users/users.service.ts

```diff
@@ -14,17 +16,48 @@ export class UsersService {
     private usersRepository: Repository<User>,
   ) {}
 
-  async create(createUserInput: CreateUserInput): Promise<User> {
+  async create(createUserInput: CreateUserInput, user: User): Promise<User> {
     try {
+      if (user.role !== Roles.ADMIN) {
```

Comment:
let's move it to Admin guard maybe ?

Reply:
I need your help in implementation because I had trouble doing it? so I left it as it is

Reply:
lest discuss it later